prow/flagutil/config.go

```diff
@@ -0,0 +1,57 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package flagutil
+
+import (
+	"flag"
+
+	"k8s.io/test-infra/prow/config"
+)
+
+// ConfigOptions holds options for interacting with Configs.
+type ConfigOptions struct {
+	ConfigPath    string
+	jobConfigPath string
+}
+
+// AddFlags injects config options into the given FlagSet.
+func (o *ConfigOptions) AddFlags(fs *flag.FlagSet) {
+	fs.StringVar(&o.ConfigPath, "config-path", "/etc/config/config.yaml", "Path to config.yaml.")
+	fs.StringVar(&o.jobConfigPath, "job-config-path", "", "Path to prow job configs.")
+}
+
+// Validate validates config options.
+func (o *ConfigOptions) Validate(dryRun bool) error {
+	return nil
```

Comment:
Why doesn't this require that the path(s) are non-nil?

Reply:
Because [`tot` allows for it to be empty][1].  I could add `ConfigOptions.RequireConfigPath` or `ConfigOptions.AllowEmptyConfigPath` or some such, but it's about as easy to check locally in the consumers.  Related previous discussion [here][2], which we resolved by adding a default (and we still only warn when `TokenPath` is unset since #9916).  How did you want me to handle this setting?

[1]: https://github.com/kubernetes/test-infra/blob/bfbc61258394e31ce8da16e204cedd5b19763a3a/prow/cmd/tot/main.go#L287
[2]: https://github.com/kubernetes/test-infra/pull/9503#discussion_r219575695

Reply:
Let's just require it for `tot`? I am not particularly interested in supporting that weird edge case and we can get more clarity on the flags if it's just required always.

/cc @cjwagner